spec/models/preserved_object_spec.rb

```diff
@@ -106,4 +106,72 @@
       end
     end
   end
+
+  describe '#create_archive_copies' do
+    let(:druid) { 'ab123cd4567' }
+    let(:current_version) { 3 }
+    let!(:po) { create(:preserved_object, druid: druid, current_version: current_version) }
+    let(:archive_ep) { Endpoint.find_by(endpoint_name: 'mock_archive1') }
+    let(:new_archive_ep) { create(:archive_endpoint, endpoint_name: 'mock_archive2') }
+    let(:archive_pcs_for_druid) do
+      PreservedCopy
+        .joins(endpoint: [:endpoint_type])
+        .where(preserved_object: po, endpoint_types: { endpoint_class: 'archive' })
+    end
+
+    it "creates pres copies that don't yet exist for the given version, but should" do
+      expect { po.create_archive_copies(current_version) }.to change {
+        Endpoint.which_need_archive_copy(druid, current_version).to_a
+      }.from([archive_ep]).to([])
+
+      expect(archive_pcs_for_druid.count).to eq 1
+
+      expect { po.create_archive_copies(current_version - 1) }.to change {
+        Endpoint.which_need_archive_copy(druid, current_version - 1).to_a
+      }.from([archive_ep]).to([])
+      expect(archive_pcs_for_druid.count).to eq 2
+
+      expect(archive_pcs_for_druid.where(version: 1).count).to eq 0
+    end
+
+    it "creates pres copies that don't yet exist for the given endpoint, but should" do
+      expect { po.create_archive_copies(current_version) }.to change {
+        Endpoint.which_need_archive_copy(druid, current_version).to_a
+      }.from([archive_ep]).to([])
+      expect(archive_pcs_for_druid.where(version: current_version).count).to eq 1
+
+      new_archive_ep.preservation_policies = [PreservationPolicy.default_policy]
+      expect { po.create_archive_copies(current_version) }.to change {
+        Endpoint.which_need_archive_copy(druid, current_version).to_a
+      }.from([new_archive_ep]).to([])
+      expect(archive_pcs_for_druid.where(version: current_version).count).to eq 2
+    end
+
```

Comment:
should we have these tests?

- does not create pres copy if it already exists
- archive copies created have status unreplicated
...

Reply:
> * does not create pres copy if it already exists

hmm... i was thinking that i'd covered that in the tests that re-run ~~`create_archive_copies`~~ `create_archive_preserved_copies` and then confirm the pres copy counts are right, but it's true that they don't test invocation on the same druid/version combo multiple times.  i will add that in #796, if you don't mind.

> * archive copies created have status unreplicated

good catch, will also do that in 796

Reply:
You can also punt on this and just merge the DB constraint.